File: sfa_mobile/lib/features/products/domain/entities/product.dart
import 'package:equatable/equatable.dart';

class Product extends Equatable {
  final int id;
  final String code;
  final String itemDescription;
  final String? printDescription;
  final int piecesPerPack;
  final String? imageUrl;
  final int? categoryId;
  final double dealerPackPrice;
  final double dealerCasePrice;
  final double mrp;

  const Product({
    required this.id,
    required this.code,
    required this.itemDescription,
    this.printDescription,
    required this.piecesPerPack,
    this.imageUrl,
    this.categoryId,
    this.dealerPackPrice = 0,
    this.dealerCasePrice = 0,
    this.mrp = 0,
  });

  @override
  List<Object?> get props => [
        id,
        code,
        itemDescription,
        printDescription,
        piecesPerPack,
        imageUrl,
        categoryId,
        dealerPackPrice,
        dealerCasePrice,
        mrp,
      ];
}

File: sfa_api/sfa_api/Features/Products/Entities/Product.cs
using sfa_api.Features.Fleets.Entities;
using sfa_api.Features.ProductCategories.Entities;

namespace sfa_api.Features.Products.Entities;

public class Product
{
    public int Id { get; set; }
    public string Code { get; set; } = string.Empty;             // Unique item code
    public string ItemDescription { get; set; } = string.Empty;  // Full display name
    public string? PrintDescription { get; set; }                 // Uppercase label for print/reports
    public int PiecesPerPack { get; set; } = 0;                  // Units per pack
    public string? ImageUrl { get; set; }                         // Storage path or URL
    public string? Remarks { get; set; }
    public int? FleetId { get; set; }
    public int? CategoryId { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsDeleted { get; set; } = false;

    // Audit fields
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    public int? CreatedBy { get; set; }
    public int? UpdatedBy { get; set; }

    // Navigation
    public Fleet? Fleet { get; set; }
    public ProductCategory? Category { get; set; }
}
